### DISCUSSION TITLE
Computer Science questions

### DISCUSSION THREAD

**Discussion by nbidler** [deepmodeling/deepmd-kit#2632] [Q&A] [answered]
Hello - I am a Computer Science master's degree student, and I am trying to help a chemistry professor with his research project. I'm less strictly interested in "getting the code to work"/the chemistry part and more interested in "tracking what makes a neural net work better"/"the inner workings."

I am trying to help optimize the training of the neural network, but I am having trouble determining what I can do without breaking the system (or writing custom code). The professor has found/chosen the relevant chemistry-related settings, which leaves me to experiment with "number of"/"nodes per" layer of the neural network. 

For example, if I wanted to see what the effect of dropout nodes on the output, I might be able to implement dropout nodes for the neural net by doing something with TensorFlow... but I don't want to make it myself if it already exists and I just haven't found it in the documentation.

Is there anything in the documentation about how DeePMD uses TensorFlow or how it works "under the hood," or should I be asking questions directly?  (i.e. is DeePMD kind of like a Graph Neural Net or no? To better narrow down how to structure the NN for my particular problem)

**Comment by njzjz** [comment, marked as answer]
You can read [our recent preprint](https://doi.org/10.48550/arXiv.2304.09409), which introduces most of things you want to know.
